src/libOpenImageIO/imagebufalgo_opencv.cpp

```diff
@@ -136,7 +137,182 @@ IplImage *
+        }
+    }
+
+    // Create an IplImage to write to.
+    CvSize size = cvSize(spec.width, spec.height);
+    IplImage *dst;
+
+    // Get the pixel data from the ImageBuf and send it to the IplImage.
+    switch (spec.format.basetype) {
+        case TypeDesc::UINT8: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_8U, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<unsigned char> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::UINT8, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        case TypeDesc::INT8: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_8S, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<signed char> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::INT8, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        case TypeDesc::UINT16: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_16U, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<unsigned int> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::UINT16, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        case TypeDesc::INT16: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_16S, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<signed int> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::INT16, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        // OpenCV doesn't have a half type, so promote to float.
+        case TypeDesc::HALF: {
```

Comment:
I think you can just more the `case TypeDesc::HALF:` to share the code block with FLOAT, no need to duplicate it like this, right?